simd/src/Kokkos_SIMD_AVX2.hpp

```diff
@@ -957,12 +957,6 @@ class const_where_expression<simd_mask<double, simd_abi::avx2_fixed_size<4>>,
       if (m_mask[lane]) mem[index[lane]] = m_value[lane];
     }
   }
-
-  friend constexpr auto const& Impl::mask<double, abi_type>(
-      const_where_expression<mask_type, value_type> const& x);
-
-  friend constexpr auto const& Impl::value<double, abi_type>(
-      const_where_expression<mask_type, value_type> const& x);
```

Comment:
Don't we need replacement `impl_get_value()` and `impl_get_mask()` functions here as well? Same for NEON simd.

Reply:
Possibly, it just doesn't seem to be needed in the tests.

Reply:
Looks like `reduce`, `hmin`, and `hmax` are not tested.

Reply:
I verified manually that we need `impl_get_value/mask` for `reduce`, `hmin`, and `hmax` and added those.
We should test those but I haven't done that here.